ios/Classes/PTPluginUtils.m

```diff
@@ -186,4 +220,174 @@ + (void)getPageCropBox:(NSNumber *)pageNumber resultToken:(FlutterResult)result
+        toolClass = [PTFreeHandHighlightCreate class];
+    } else if ([toolMode isEqualToString:PTAnnotationCreateRubberStampToolKey]) {
+        toolClass = [PTRubberStampCreate class];
+        
+    }
+    
+    if (toolClass) {
+        PTTool *tool = [docVC.toolManager changeTool:toolClass];
+        
+        tool.backToPanToolAfterUse = !continuousAnnotEditing;
+
+        if ([tool isKindOfClass:[PTFreeHandCreate class]]
+            && ![tool isKindOfClass:[PTFreeHandHighlightCreate class]]) {
+            ((PTFreeHandCreate *)tool).multistrokeMode = continuousAnnotEditing;
+        }
+    }
+    
+    result(nil);
+}
+
++ (void)setFlagForFields:(NSArray <NSString *> *)fieldNames flag:(NSNumber *)flag flagValue:(bool)flagValue resultToken:(FlutterResult)result documentViewController:(PTDocumentViewController *)docVC
+{
+    if(docVC.document == Nil)
+    {
+        // something is wrong, no document.
+        NSLog(@"Error: The document view controller has no document.");
+        return;
+    }
+    
+    PTPDFViewCtrl *pdfViewCtrl = docVC.pdfViewCtrl;
+    PTFieldFlag fieldFlag = (PTFieldFlag)flag.intValue;
+    NSError *error;
+  
+    [pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+        for (NSString *fieldName in fieldNames) {
+            PTField *field = [doc GetField:fieldName];
+            if ([field IsValid]) {
+                [field SetFlag:fieldFlag value:flagValue];
+                [pdfViewCtrl UpdateWithField:field];
+            }
+        }
+    } error:&error];
+   
+    if (error) {
+        @throw [NSException exceptionWithName:NSGenericException reason:error.localizedFailureReason userInfo:error.userInfo];
+    }
+    
+    result(nil);
+}
+
++ (void)setValueForFields:(NSString *)fieldWithValuesString resultToken:(FlutterResult)result documentViewController:(PTDocumentViewController *)docVC
+{
+    NSArray *fieldWithValues = [PTPluginUtils PT_idAsNSArray:[PTPluginUtils PT_JSONStringToId:fieldWithValuesString]];
+    if(docVC.document == Nil)
+    {
+        // something is wrong, no document.
+        NSLog(@"Error: The document view controller has no document.");
+        return;
+    }
+    
+    PTPDFViewCtrl *pdfViewCtrl = docVC.pdfViewCtrl;
+    NSError *error;
+    
+    [pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+        
+        for (NSDictionary *fieldWithValue in fieldWithValues) {
+            NSString *fieldName = [PTPluginUtils PT_idAsNSString:fieldWithValue[PTFieldNameKey]];
+            id fieldValue = fieldWithValue[PTFieldValueKey];
+            PTField *field = [doc GetField:fieldName];
+            
+            if ([field IsValid]) {
+                [PTPluginUtils setFieldValue:field value:fieldValue pdfViewCtrl:pdfViewCtrl];
+            }
+        }
+        
+    } error:&error];
+    
+    if (error) {
+        @throw [NSException exceptionWithName:NSGenericException reason:error.localizedFailureReason userInfo:error.userInfo];
```

Comment:
Do we really want to be throwing exceptions from here? Does something catch this and pass it back to Flutter nicely?